src/webui/api/appcontroller.cpp

```diff
@@ -651,12 +649,23 @@ void AppController::setPreferencesAction()
     }
     // Current network interface address
     if (hasKey("current_interface_address")) {
-        const QHostAddress ifaceAddress {it.value().toString().trimmed()};
-        session->setNetworkInterfaceAddress(ifaceAddress.isNull() ? QString {} : ifaceAddress.toString());
+        const QString value = it.value().toString().trimmed();
+        bool ok = false;
+        uint option = value.toUInt(&ok);
```

Comment:
On a second thought, the original code of this block should be preserved and should let webUI return the ip address string because now the API doesn't make sense.

Reply:
And how will the Webui user select "I want all IPv4" addresses for the network interface? (vs all address vs all ipv6 addresses vs specific address)

Reply:
PS: The values here mirror the ones chosen for GUI combobox. They represent the position of the 3 top items in the combobox list.

Reply:
>And how will the Webui user select "I want all IPv4" addresses for the network interface? 

I guess some modification in js code is required, that is to mimic the new combobox options in GUI.

>The values here mirror the ones chosen for GUI combobox. They represent the position of the 3 top items in the combobox list.

You see the api key here is `current_interface_address`, so unless you document explicitly this magic value in WebAPI document, none of the users of WebAPI could guess what `1` or `2` means and not to mention this semantic change warrants a webapi version bump.

Reply:
>so unless you document explicitly these magic values in WebAPI document

I can do that. Just point me to the correct place.

>warrants a webapi version bump.

The webapi major version bump will happen anyway because I remove the `listen_on_ipv6_address` key

Reply:
@Piccirello 
Any comment on this discussion? https://github.com/qbittorrent/qBittorrent/pull/11592#discussion_r355530995

Reply:
Guys, I really don't understand why you can't accept it as it processed by app core (i.e. `0.0.0.0` and `::`)? Why all these "magic" values of separate type?

Reply:
In this case `0.0.0.0` and `::` will be treated as magic values from the UI perspective. The UI must not show them as `0.0.0.0` and `::` but as `All IPv4` and `All IPv6` respectively. In this context they have special meaning. And we need another value that will indicate `All addresses` (could be an empty string).

So it is a choice on the set of magic numbers. But in this case, don't we run the risk of web devs ignoring these special cases if we just send `0.0.0.0` and `::` ? (sending 0/1/2) will force them to take a better look (and their users will point that out)

In any case, I am no expert in designing outward facing API nor do I care much about the webapi stuff.
If you find a consensus on how it should be represented via the WebAPI, I'll happily oblige.

Reply:
>Guys, I really don't understand why you can't accept it as it processed by app core (i.e. 0.0.0.0 and ::)? 

That is what I meant in the first post.

>And we need another value that will indicate All addresses (could be an empty string).

An empty string for all address is fine for me or even better IMO: `0.0.0.0,::`

>But in this case, don't we run the risk of web devs ignoring these special cases if we just send 0.0.0.0 and :: ? 

From OS API perspective, these 2 values are not special, they just binds to any IPv4/IPv6 address respectively. And any one who ever worked with network sockets will recognize these values. Or do you have other *special* meaning?

>In any case, I am no expert in designing outward facing API nor do I care much about the webapi stuff.

I'm sorry to have to say this, you're touching webapi here and one mistake here is going to cause inconveniences for webAPI devs/users for some period.

>If you find a consensus on how it should be represented via the WebAPI, I'll happily oblige.

`current_interface_address` can only expect an IPV4/IPv6 address string or an empty string.

As for WebUI, (warning, pseudo code blow)
Here need to add an `if (item === '::') // add "all ipv6" option` to populate the entries:
https://github.com/qbittorrent/qBittorrent/blob/975b44d05ffceb147523cc99f878ce8a5a43920e/src/webui/www/private/views/preferences.html#L1442

And here needs to modified to get the ip address value, probably something like: `if ($('optionalIPAddressToBind').getProperty('value') === 'all ipv6') // set to "::"`
https://github.com/qbittorrent/qBittorrent/blob/975b44d05ffceb147523cc99f878ce8a5a43920e/src/webui/www/private/views/preferences.html#L2095

Reply:
>So it is a choice on the set of magic numbers. But in this case, don't we run the risk of web devs ignoring these special cases if we just send `0.0.0.0` and `::`?

In any case it's a bad idea to blindly show any received data. UI level should make postprocessing and use appropriate representation. When magic values are documented there is no difference between numbers and "special" IPs. But as I said before "special" IPs are better for readers (for debugging purposes). So even some "stupid" UI shows it as is its better than see some really meaningless numbers.

P.S. Even if the others vote for the numbers and you decide to use them, you don't need to convert anything here, just compare them in string form.